### PR TITLE
testutil/promrated: update promrated querying

### PR DESCRIPTION
**Pull request by LukeHackett12**
Promrated is needed to fulfill a slightly different purpose these days, refactoring to reflect that and adding a needed endpoint.

category: refactor
ticket: #2843

### CODE REVIEW

**Comment by pinebit** [comment]
Pay attention to linter issues:
```
  Error: const `promQuery` is unused (unused)
  Error: type `validator` is unused (unused)
  Error: func `getValidators` is unused (unused)
  Error: func `parseValidators` is unused (unused)
```
You can run `pre-commit` locally to see them before committing to save time.

**Comment by LukeHackett12** [comment]
pre commit passed locally 😞 , think it should be fixed now!

File: testutil/promrated/promrated.go [hunk 1]
```diff
@@ -112,24 +78,30 @@ func reportMetrics(ctx context.Context, config Config) {
...
-// contains checks if array contains a string s.
-func contains(arr []string, s string) bool {
-	result := false
-	for _, x := range arr {
-		if x == s {
-			result = true
-			break
+			stats, err = getNodeOperatorStatistics(ctx, config.RatedEndpoint, config.RatedAuth, nodeOperator, network)
+			if err != nil {
+				log.Error(ctx, "Getting node operator statistics", err, z.Str("network", network), z.Str("node_operator", "Lido"))
```
Comment (pinebit) [inline testutil/promrated/promrated.go:91]: Why is this fixed to Lido?

File: testutil/promrated/metrics.go [hunk 2]
```diff
@@ -9,44 +9,7 @@ import (
...
-		Name:      "validator_proposer_effectiveness",
-		Help:      "Proposer effectiveness of a validation key.",
-	}, validatorLabels)
-
-	effectiveness = promauto.NewGaugeVec(prometheus.GaugeOpts{
-		Namespace: "promrated",
-		Name:      "validator_effectiveness",
-		Help:      "Effectiveness of a validation key.",
-	}, validatorLabels)
+	networkLabels = []string{"cluster_network", "node_operator"}
```
Comment (pinebit) [inline testutil/promrated/metrics.go:12]: These literals are repeated explicitly in the code below. I suggest to introduce string constants for these.

File: testutil/promrated/promrated/main.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func bindPromratedFlag(flags *pflag.FlagSet, config *promrated.Config) {
 	flags.StringVar(&config.RatedEndpoint, "rated-endpoint", "https://api.rated.network", "Rated API endpoint to poll for validator metrics.")
 	flags.StringVar(&config.RatedAuth, "rated-auth-token", "token", "[REQUIRED] Token for Rated API.")
 	flags.StringVar(&config.MonitoringAddr, "monitoring-address", "127.0.0.1:9100", "Listening address (ip and port) for the prometheus monitoring http server.")
-	flags.StringVar(&config.PromEndpoint, "prom-endpoint", "https://vm.monitoring.gcp.obol.tech/query", "Endpoint for VMetrics Prometheus API.")
 	flags.StringVar(&config.PromAuth, "prom-auth-token", "token", "[REQUIRED] Token for VMetrics Promtetheus API.")
 	flags.StringSliceVar(&config.Networks, "networks", nil, "Comma separated list of one or networks to monitor.")
+	flags.StringSliceVar(&config.Networks, "node-operators", nil, "Comma separated list of one or node operators to monitor.")
```
Comment (pinebit) [inline testutil/promrated/promrated/main.go:47]: Copy-paste error!

**Review by pinebit — CHANGES_REQUESTED** [review]
Please fix the copy-paste error 
`flags.StringSliceVar(&config.Networks, "node-operators", nil, "Comma separated list of one or node operators to monitor.")`

File: testutil/promrated/promrated/main.go (hunk 3 above)
Comment (pinebit) [reply testutil/promrated/promrated/main.go:47]: Also, do we need any validation for these? If the node operators list is not provided then the labels won't be populated?

Comment (LukeHackett12) [reply testutil/promrated/promrated/main.go:47]: that would be the intended behaviour, I think that this would be OK unless you have a strong objection?

**Review by pinebit — APPROVED** [review]
LGTM, let's get at least one more approval from Protocol.

File: testutil/promrated/promrated.go [hunk 4]
```diff
@@ -42,7 +43,7 @@ func Run(ctx context.Context, config Config) error {
 		serverErr <- serveMonitoring(config.MonitoringAddr, promRegistry)
 	}()
 
-	ticker := time.NewTicker(12 * time.Hour)
+	ticker := time.NewTicker(24 * time.Hour)
```
Comment (dB2510) [inline testutil/promrated/promrated.go:46]: What's the reason behind updating this ticker? to reduce number of queries per day? A comment would be great for the same.

File: testutil/promrated/rated.go [hunk 5]
```diff
@@ -18,61 +18,53 @@ import (
...
-	ValidatorEffectiveness float64 `json:"validatorEffectiveness"`
-}
-
 type networkEffectivenessData struct {
 	AvgUptime              float64 `json:"avgUptime"`
 	AvgCorrectness         float64 `json:"avgCorrectness"`
 	AvgInclusionDelay      float64 `json:"avgInclusionDelay"`
 	ValidatorEffectiveness float64 `json:"avgValidatorEffectiveness"`
+	ProposerEffectiveness  float64 `json:"avgProposerEffectiveness"`
+	AttesterEffectiveness  float64 `json:"avgAttesterEffectiveness"`
```
Comment (dB2510) [inline testutil/promrated/rated.go:27]: nit: 
```suggestion
	AvgValidatorEffectiveness float64 `json:"avgValidatorEffectiveness"`
	AvgProposerEffectiveness  float64 `json:"avgProposerEffectiveness"`
	AvgAttesterEffectiveness  float64 `json:"avgAttesterEffectiveness"`
```

File: testutil/promrated/rated.go [hunk 6]
```diff
@@ -126,38 +118,37 @@ func queryRatedAPI(ctx context.Context, url *url.URL, ratedAuth string, network
...
 	}
 
-	return result.Data[0], nil
+	return result[0], nil
 }
 
-// parseNetworkMetrics reads the network rated response and returns the network effectiveness data.
-func parseNetworkMetrics(body []byte) (networkEffectivenessData, error) {
-	var result []networkEffectivenessData
+func parseNodeOperatorMetrics(body []byte) (networkEffectivenessData, error) {
```
Comment (dB2510) [inline testutil/promrated/rated.go:137]: Add godoc

File: testutil/promrated/rated_internal_test.go [hunk 7]
```diff
@@ -14,69 +14,58 @@ import (
 	"github.com/obolnetwork/charon/testutil"
 )
 
-//go:generate go test . -update -clean
-
-func TestGetValidatorStatistics(t *testing.T) {
+func TestGetNetworkStatistics(t *testing.T) {
```
Comment (dB2510) [inline testutil/promrated/rated_internal_test.go:17]: add the go generate line

**Review by dB2510 — APPROVED** [review]

**Review by xenowits — APPROVED** [review]

**Pull request merged**